handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/LoadBalancer.java

```diff
@@ -65,7 +65,19 @@ public ServiceCombServer chooseServer(Invocation invocation) {
         break;
       }
     }
-    return rule.choose(servers, invocation);
+    ServiceCombServer server = rule.choose(servers, invocation);
+    if (null == server) {
+      return null;
+    }
+    ServiceCombServerStats serverStats = ServiceCombLoadBalancerStats.INSTANCE.getServiceCombServerStats(server);
+    if (ServiceCombServerStatus.ISOLATED == serverStats.getStatus()
+        && serverStats.markStatusAtomically(ServiceCombServerStatus.ISOLATED, ServiceCombServerStatus.TRYING)) {
```

Comment:
and maybe will have ABA problem?

multiple requests run into line 73
one go with success at line 74, and failed very quickly at last, will mark back to ISOLATED
then other requests run to line 74, the markStatusAtomically will still be success, so will try multiple times 

maybe detect the target instance maybe can work by PING mechanism is better?
only PING successed instance allow to TRY in ISOLATED status?
and customer can customize PING mechanism to detect real business status.

Reply:
What about adding a status `LOCKED_FOR_TRYING`? The `ISOLATED` instances are picked out and tansfered to `LOCKED_FOR_TRYING` atomically. And in LoadBalancer, every `LOCKED_FOR_TRYING` instances not selected are transfered back to `ISOLATED`, waiting for the next invocation.